### PR TITLE
add DeFi Protocol Tokens

### PR DESCRIPTION
**Pull request by aranhaagency**
#307 
- [x] Compound - https://compound.finance/developers

- [x] Fulcrum - https://github.com/bZxNetwork/fulcrum_ui/blob/development/packages/mmaker-wrappers/assets/artifacts/tokenList.ts
- [x] Uniswap - https://github.com/Uniswap/uniswap-frontend/blob/5bcbe0ee8fb4a1524b43ab34b495773c129d4765/src/contexts/Tokens.js
- [x] Synthetix - https://developer.synthetix.io/api/docs/deployed-contracts.html
- [x] AAVE Protocol

### CODE REVIEW

**Comment by ligi** [comment]
I saw it is WIP - still left the comment so that the error does not propagate further

**Comment by aranhaagency** [comment]
I fixed this Json now, thanks for the quick feedback.

File: tokens/eth/0x692a2b8bE7E166D6EE93B22A4B8B351e5d444339.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dsKNC2x",
+    "address": "0x30BB2D30B3A3A3F4943f81d460b45D2dac5735DF",
```
Comment (ligi) [inline tokens/eth/0x692a2b8bE7E166D6EE93B22A4B8B351e5d444339.json]: This address does not match the filename

File: tokens/eth/0xA62D9d68674AaAc7Cb31E8c0515Dd3A47D897363.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "MANA",
+    "address": "0xA62D9d68674AaAc7Cb31E8c0515Dd3A47D897363",
+    "decimals": 18,
+    "name": "Aave Interest bearing KNC",
```
Comment (blurpesec) [inline tokens/eth/0xA62D9d68674AaAc7Cb31E8c0515Dd3A47D897363.json]: This name, and the symbol are somewhat misaligned

Comment (aranhaagency) [reply tokens/eth/0xA62D9d68674AaAc7Cb31E8c0515Dd3A47D897363.json]: Thanks @blurpesec, I will fixed this.

**Comment by aranhaagency** [comment]
hey @ligi  I fixed.

File: tokens/eth/0xff8Be4B22CeDC440591dcB1E641EB2a0dd9d25A5.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "URAC",
+    "address": "00xff8Be4B22CeDC440591dcB1E641EB2a0dd9d25A5",
```
Comment (ligi) [inline tokens/eth/0xff8Be4B22CeDC440591dcB1E641EB2a0dd9d25A5.json]: This address is not correct

**Comment by aranhaagency** [comment]
Hi @ligi I didn't quite understand, do you want me to add the type of address in json?

**Comment by ligi** [comment]
PS: if you run the gradle script in the repo - e.g. via ./gradlew run you can find these problems yourself

**Comment by aranhaagency** [comment]
@ligi Thanks, I fixed.

**Comment by ligi** [comment]
Thanks! The CI script is happy now - will start to have a deeper look now - can take a bit with this amount of tokens

**Comment by aranhaagency** [comment]
@ligi okay, anything like that to call.

File: tokens/eth/0xde37db4269C6dfD4c81B9A11400d1BcBEeC06515.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "usZRX4x",
+    "address": "0xde37db4269C6dfD4c81B9A11400d1BcBEeC06515",
+    "decimals": 8,
```
Comment (ligi) [inline tokens/eth/0xde37db4269C6dfD4c81B9A11400d1BcBEeC06515.json]: the contract reports 18 and not 8

**Comment by ligi** [comment]
Sorry for the delay with the review - but these are just so many tokens and I need to check properly. Added new scripts for this and found one more problem with it.

File: tokens/eth/0xde37db4269C6dfD4c81B9A11400d1BcBEeC06515.json (hunk 4 above)
Comment (aranhaagency) [reply tokens/eth/0xde37db4269C6dfD4c81B9A11400d1BcBEeC06515.json]: Thanks I fixed.

**Comment by aranhaagency** [comment]
no problem @ligi , we have to check everything.

File: tokens/eth/0x289e9a4674663decEE54f781AaDE5327304A32f8.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "SynthsUSD",
```
Comment (ligi) [inline tokens/eth/0x289e9a4674663decEE54f781AaDE5327304A32f8.json]: symbol reported from contract is sUSD

File: tokens/eth/0x485B627f43E378417125b49Ed95CAc28b78339C4.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "TUSD",
+    "address": "0x485B627f43E378417125b49Ed95CAc28b78339C4",
```
Comment (ligi) [inline tokens/eth/0x485B627f43E378417125b49Ed95CAc28b78339C4.json]: symbol from contract is dsKNC

File: tokens/eth/0x46824bFAaFd049fB0Af9a45159A88e595Bbbb9f7.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "SynthsLINK",
```
Comment (ligi) [inline tokens/eth/0x46824bFAaFd049fB0Af9a45159A88e595Bbbb9f7.json]: sLINK

File: tokens/eth/0xd7adF1b5E31D1C40E08F16a2095338ce3aA8f2Fc.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "SynthiXRP",
```
Comment (ligi) [inline tokens/eth/0xd7adF1b5E31D1C40E08F16a2095338ce3aA8f2Fc.json]: iXRP

File: tokens/eth/0xD199A3F8a65fD0C80A47718cB5E2D1C9f3729bC6.json [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "uLBAT2x",
+    "address": "0xD199A3F8a65fD0C80A47718cB5E2D1C9f3729bC6",
+    "decimals": 18,
```
Comment (ligi) [inline tokens/eth/0xD199A3F8a65fD0C80A47718cB5E2D1C9f3729bC6.json]: contract reports 6

File: tokens/eth/0xC5Bfbc63dc8D36E81434e93e0ee097999879d7F4.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "SynthiDEFI",
```
Comment (ligi) [inline tokens/eth/0xC5Bfbc63dc8D36E81434e93e0ee097999879d7F4.json]: iDEFI

File: tokens/eth/0x4F4D523C69a47c5C3ef06c53ec64801f11a884Dd.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dsZRX",
```
Comment (ligi) [inline tokens/eth/0x4F4D523C69a47c5C3ef06c53ec64801f11a884Dd.json]: dLWBTC4x

File: tokens/eth/0xbd408612BACccbF14FF26Ca0dEF859FacC3673bD.json [hunk 12]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "uLWBTC2x",
+    "address": "0xbd408612BACccbF14FF26Ca0dEF859FacC3673bD",
+    "decimals": 18,
```
Comment (ligi) [inline tokens/eth/0xbd408612BACccbF14FF26Ca0dEF859FacC3673bD.json]: contract reports 6

**Comment by ligi** [comment]
with the last force push you added a lot of files that should not be checked in

**Comment by aranhaagency** [comment]
Wow, I'm going to update the pull request sorry.

**Comment by aranhaagency** [comment]
@ligi I removed unnecessary files from the pull request

File: tokens/eth/0x692a2b8bE7E166D6EE93B22A4B8B351e5d444339.json [hunk 13]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dsKNC2x",
+    "address": "0x692a2b8bE7E166D6EE93B22A4B8B351e5d444339",
```
Comment (ligi) [inline tokens/eth/0x692a2b8bE7E166D6EE93B22A4B8B351e5d444339.json]: dsKNC

File: tokens/eth/0x1287969821F9160C1aF516Af0ff18db2903386aD.json [hunk 14]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "uLETH4x",
+    "address": "0x1287969821F9160C1aF516Af0ff18db2903386aD",
+    "decimals": 18,
```
Comment (ligi) [inline tokens/eth/0x1287969821F9160C1aF516Af0ff18db2903386aD.json]: 6

File: tokens/eth/0x240fe85447A878f51A74A5dC0b644b4A72587839.json [hunk 15]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "usETH",
```
Comment (ligi) [inline tokens/eth/0x240fe85447A878f51A74A5dC0b644b4A72587839.json]: dLREP4x

File: tokens/eth/0x43baf2ec0C4236B661a84FD40EC076546E3Bb9Fd.json [hunk 16]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "usZRX",
+    "address": "0x43baf2ec0C4236B661a84FD40EC076546E3Bb9Fd",
+    "decimals": 8,
```
Comment (ligi) [inline tokens/eth/0x43baf2ec0C4236B661a84FD40EC076546E3Bb9Fd.json]: 18

File: tokens/eth/0x2C537E5624e4af88A7ae4060C022609376C8D0EB.json [hunk 17]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "BiLira",
```
Comment (ligi) [inline tokens/eth/0x2C537E5624e4af88A7ae4060C022609376C8D0EB.json]: TRYB

File: tokens/eth/0x92bA756169aA6Ff82a3d6ae8e4456A5883182Fa3.json [hunk 18]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dLLINK3x_v2",
```
Comment (ligi) [inline tokens/eth/0x92bA756169aA6Ff82a3d6ae8e4456A5883182Fa3.json]: dLLINK3x

File: tokens/eth/0x937EFce07594d14f878650d5B637f022952FAe86.json [hunk 19]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dsLINK4x_v2",
```
Comment (ligi) [inline tokens/eth/0x937EFce07594d14f878650d5B637f022952FAe86.json]: dsLINK4x

File: tokens/eth/0x1abb24C9c806eAA71750Fa05C4009bd32e65ebad.json [hunk 20]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "dLLINK2x_v2",
```
Comment (ligi) [inline tokens/eth/0x1abb24C9c806eAA71750Fa05C4009bd32e65ebad.json]: dLLINK2x

File: tokens/eth/0x187234E7a0C64dCEF6176a534ef1E9e627D9aDC8.json [hunk 21]
```diff
@@ -0,0 +1,34 @@
+{
+    "symbol": "uLREP2x",
+    "address": "0x187234E7a0C64dCEF6176a534ef1E9e627D9aDC8",
+    "decimals": 18,
```
Comment (ligi) [inline tokens/eth/0x187234E7a0C64dCEF6176a534ef1E9e627D9aDC8.json]: 6

**Comment by aranhaagency** [comment]
Hi @ligi, I run command `./gradlew run`  everything is ok now.

**Comment by aranhaagency** [comment]
> yay - great - the CI is happy and the PR looks generally good. So we are close to merging here 🎉
> One last thing I do not like about it is the logo section that have a width and a height but no content. This makes no sense to me (and I should make it part of the CI checks) - when this is done I am happy to merge - thanks for your work!

I Will fixed this.

**Comment by aranhaagency** [comment]
@ligi done!

**Pull request merged**